features/netsocket/SocketAddress.cpp

```diff
@@ -250,34 +252,38 @@ uint16_t SocketAddress::get_port() const
 
 SocketAddress::operator bool() const
 {
-    int count = 0;
     if (_addr.version == NSAPI_IPv4) {
-        count = NSAPI_IPv4_BYTES;
-    } else if (_addr.version == NSAPI_IPv6) {
-        count = NSAPI_IPv6_BYTES;
-    }
+        for (int i = 0; i < NSAPI_IPv4_BYTES; i++) {
+            if (_addr.bytes[i]) {
+                return true;
+            }
+        }
 
-    for (int i = 0; i < count; i++) {
-        if (_addr.bytes[i]) {
-            return true;
+        return false;
+    } else if (_addr.version == NSAPI_IPv6) {
+        for (int i = 0; i < NSAPI_IPv6_BYTES; i++) {
+            if (_addr.bytes[i]) {
+                return true;
+            }
         }
-    }
 
-    return false;
+        return false;
+    } else {
+        return false;
+    }
 }
 
 bool operator==(const SocketAddress &a, const SocketAddress &b)
 {
-    int count = 0;
-    if (a._addr.version == NSAPI_IPv4 && b._addr.version == NSAPI_IPv4) {
-        count = NSAPI_IPv4_BYTES;
+    if (a._addr.version == NSAPI_UNSPEC && b._addr.version == NSAPI_UNSPEC) {
```

Comment:
Could check different versions first and return false.

But should UNSPEC be equal to IPv4 0.0.0.0?